src/main/java/network/brightspots/rcv/GuiConfigController.java

```diff
@@ -793,27 +795,27 @@ private void loadConfig(ContestConfig config) {
       tableViewCandidates.setItems(FXCollections.observableArrayList(rawConfig.candidates));
     }
 
-    ContestRules rules = rawConfig.rules;
     choiceTiebreakMode.setValue(config.getTiebreakMode());
     choiceOvervoteRule.setValue(config.getOvervoteRule());
-    setTextFieldToInteger(textFieldNumberOfWinners, rules.numberOfWinners);
+    setTextFieldToInteger(textFieldNumberOfWinners, config.getNumberOfWinners());
     setTextFieldToInteger(
-        textFieldDecimalPlacesForVoteArithmetic, rules.decimalPlacesForVoteArithmetic);
-    setTextFieldToInteger(textFieldMinimumVoteThreshold, rules.minimumVoteThreshold);
-    setTextFieldToInteger(textFieldMaxSkippedRanksAllowed, rules.maxSkippedRanksAllowed);
-    setTextFieldToInteger(textFieldMaxRankingsAllowed, rules.maxRankingsAllowed);
-    textFieldOvervoteLabel.setText(rules.overvoteLabel);
-    textFieldUndervoteLabel.setText(rules.undervoteLabel);
-    textFieldUndeclaredWriteInLabel.setText(rules.undeclaredWriteInLabel);
-    textFieldRulesDescription.setText(rules.rulesDescription);
-    checkBoxSequentialMultiSeat.setSelected(rules.sequentialMultiSeat);
-    checkBoxBottomsUpMultiSeat.setSelected(rules.bottomsUpMultiSeat);
-    checkBoxNonIntegerWinningThreshold.setSelected(rules.nonIntegerWinningThreshold);
-    checkBoxHareQuota.setSelected(rules.hareQuota);
-    checkBoxBatchElimination.setSelected(rules.batchElimination);
-    checkBoxContinueUntilTwoCandidatesRemain.setSelected(rules.continueUntilTwoCandidatesRemain);
-    checkBoxExhaustOnDuplicateCandidate.setSelected(rules.exhaustOnDuplicateCandidate);
-    checkBoxTreatBlankAsUndeclaredWriteIn.setSelected(rules.treatBlankAsUndeclaredWriteIn);
+        textFieldDecimalPlacesForVoteArithmetic, config.getDecimalPlacesForVoteArithmetic());
+    setTextFieldToInteger(textFieldMinimumVoteThreshold, config.getMinimumVoteThreshold().intValue());
+    setTextFieldToInteger(textFieldMaxSkippedRanksAllowed, config.getMaxSkippedRanksAllowed());
+    // use raw value for storage / persistence as it can be null
+    setTextFieldToInteger(textFieldMaxRankingsAllowed, config.rawConfig.rules.maxRankingsAllowed);
```

Comment:
I introduced the special logic for this field originally, but I'm reconsidering it now. I don't really like that we have to handle this one field differently from all the others here. If we have to read from the config instead of the raw config here (see my other comment), maybe we should push the special logic (fall back to numDeclaredCandidates when this is null) into the callsites? There are only a few.